Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -189,14 +193,47 @@ void LoadPSIMuonBin::exec() {
 
   // If timeZero is bigger than the largest bin assume it refers to a bin's
   // value
+  double absTimeZero = timeZero;
   if (timeZero > largestBinValue) {
-    setProperty("TimeZero",
-                outputWorkspace->x(0)[static_cast<int>(std::floor(timeZero))]);
-  } else {
-    setProperty("TimeZero", timeZero);
+    absTimeZero = outputWorkspace->x(0)[static_cast<int>(std::floor(timeZero))];
+  }
+  setProperty("TimeZero", absTimeZero);
+
+  auto firstGoodDataSpecIndex = static_cast<int>(
+      *std::max_element(m_header.firstGood, m_header.firstGood + 16));
+
+  setProperty("FirstGoodData", outputWorkspace->x(0)[firstGoodDataSpecIndex]);
+
+  // Time zero is when the pulse starts.
+  // The pulse should be at t=0 to be like ISIS data
+  // manually offset the data
+  for (auto specNum = 0u; specNum < m_histograms.size(); ++specNum) {
+    auto xData = outputWorkspace->mutableX(specNum);
+    for (size_t j = 0; j < xData.size(); j++) {
+      xData[j] = xData[j] - absTimeZero;
+    }
+    outputWorkspace->mutableX(specNum) = xData;
   }
 }
 
+void LoadPSIMuonBin::makeDeadTimeTable(const size_t &numSpec) {
+  if (getPropertyValue("DeadTimeTable").empty())
+    return;
+  Mantid::DataObjects::TableWorkspace_sptr deadTimeTable =
+      boost::dynamic_pointer_cast<Mantid::DataObjects::TableWorkspace>(
+          Mantid::API::WorkspaceFactory::Instance().createTable(
+              "TableWorkspace"));
+
```

Comment:
Can you add an assert here in case a future developer ever causes this dynamic pointer cast to fail
`assert(deadTimeTable);`